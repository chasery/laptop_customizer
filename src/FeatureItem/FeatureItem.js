import React from "react";
import { USCurrencyFormat } from "../USCurrencyFormat";
import slugify from "slugify";
import "./FeatureItem.css";

class FeatureItem extends React.Component {
  render() {
    const { feature, item, itemHash, selected, updateFeature } = this.props;

    return (
      <div className="feature__item">
        <input
          type="radio"
          id={itemHash}
          className="feature__option"
          name={feature ? slugify(feature) : ""}
          checked={item.name === selected[feature].name}
          onChange={(e) => updateFeature(feature, item)}
        />
        <label htmlFor={itemHash} className="feature__label">
          {item.name} ({USCurrencyFormat.format(item.cost)})
        </label>
      </div>
    );
  }
}

FeatureItem.defaultProps = {
  feature: "",
  selected: {
    "": {
      name: "",
      cost: null,
    },
  },
  item: {
    name: "",
    cost: null,
  },
};

export default FeatureItem;
