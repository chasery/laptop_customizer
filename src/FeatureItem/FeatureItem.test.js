import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeatureItem from "./FeatureItem";

describe("FeatureItem component", () => {
  it("renders without errors", () => {
    const wrapper = shallow(<FeatureItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
