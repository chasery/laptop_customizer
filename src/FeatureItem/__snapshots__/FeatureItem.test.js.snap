// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureItem component renders without errors 1`] = `
<div
  className="feature__item"
>
  <input
    checked={true}
    className="feature__option"
    name=""
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
  >
     (
    $0.00
    )
  </label>
</div>
`;
